uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Test/Fabricator/Document.php

```diff
@@ -0,0 +1,75 @@
+<?php
+use CRM_HRCore_Test_Fabricator_Contact as ContactFbricator;
+
+/**
+ * Documents Fabricator class
+ */
+class CRM_Tasksassignments_Test_Fabricator_Document {
+  private static $defaultParams = [];
+
+  private static function setDefaultParameters() {
+    $contact = ContactFbricator::fabricate();
+
+    self::$defaultParams = [
+      'activity_type_id' => self::getTypeID(),
+      'activity_date_time' => date('Y-m-d'),
+      'status_id' => CRM_Tasksassignments_BAO_Document::STATUS_AWAITING_UPLOAD,
+      'priority_id' => self::getPriorityID(),
+      'source_contact_id' => $contact['id'],
+      'target_contact_id' => [$contact['id']],
+      'assignee_contact_id' => [$contact['id']],
+    ];
+  }
+
+  public static function fabricate($params) {
+    self::setDefaultParameters();
+    return CRM_Tasksassignments_BAO_Document::create(array_merge(self::$defaultParams, $params));
+  }
+
+  public static function fabricateWithAPI($params) {
+    self::setDefaultParameters();
+    $result = civicrm_api3(
+      'Document',
+      'create',
+      array_merge(self::$defaultParams, $params)
+    );
+
+    return array_shift($result['values']);
+  }
+
+  public static function getCustomFieldName($fieldName) {

```

Comment:
this has nothing to do with document fabrication , I think you can move it to the test class that use it as private method and give it a more descriptive name